### DISCUSSION TITLE
[Deploy to GCP] "ERROR: failed to solve: cannot copy to non-directory" when Build the Docker image

### DISCUSSION THREAD

**Discussion by Elayenkee** [serverpod/serverpod#1785] [Q&A] [answered]
Hello everyone,

I'm asking for your help.
I'm trying to follow this documentation : https://docs.serverpod.dev/deployments/deploying-to-gce-terraform
I'm not familiar at all with everything like hosting, domain, server, etc..

I share to you what is happening when i follow this step : https://docs.serverpod.dev/deployments/deploying-to-gce-terraform#deploy-your-docker-container

Can anyone try to understand and guide me to the resolution of this problem ?

Thank you very much
![2024-01-19 13_10_19-Deploy to GCP · Elayenkee_Dungeoners@73f52cc](https://github.com/serverpod/serverpod/assets/48598144/01d86c18-4a51-4ee5-9410-a85ee4cf2c3d)

**Comment by miraieyes** [comment, marked as answer]
Hi there,

I've encountered the same problem but it can easily be solved by using the following docker config for 1.2.0

FROM dart:3.0 AS build

WORKDIR /app
COPY . .

RUN dart pub get
RUN dart compile exe bin/main.dart -o bin/main

FROM busybox:1.34.1

ENV runmode=development
ENV serverid=default
ENV logging=normal
ENV role=monolith

COPY --from=build /runtime/etc /etc
COPY --from=build /runtime/lib /lib
COPY --from=build /runtime/lib64 /lib64
COPY --from=build /runtime/usr /usr
COPY --from=build /app/bin/main /app/bin/main
COPY --from=build /app/config/ config/
#COPY --from=build /app/generated/ generated/
COPY --from=build /app/web/ web/

EXPOSE 8080
EXPOSE 8081
EXPOSE 8082

CMD app/bin/main --mode $runmode --server-id $serverid --logging $logging --role $role


Let me know if it did the job

**Reply by Elayenkee** [reply]
Good !

The job worked :)

Now I can go on the next steps of the tutorial and hope it will all work.

**Reply by miraieyes** [reply]
Great and Good luck!

**Comment by Tokotuu** [comment]
I faced the same issue and I solved it by updating the Dockerfile with the one from Serverpod 2.0 :

FROM dart:3.4.1 AS build

WORKDIR /app
COPY . .

RUN dart pub get
RUN dart compile exe bin/main.dart -o bin/server

FROM alpine:latest

ENV runmode=production
ENV serverid=default
ENV logging=normal
ENV role=monolith

COPY --from=build /runtime/ /
COPY --from=build /app/bin/server server
COPY --from=build /app/config/ config/
COPY --from=build /app/web/ web/

EXPOSE 8080
EXPOSE 8081
EXPOSE 8082

ENTRYPOINT ./server --mode=$runmode --server-id=$serverid --logging=$logging --role=$role
